core-java-modules/core-java-string-operations-7/src/main/java/com/baeldung/capitalizefirstcharactereachword/CapitalizeFirstCharacterEachWordUtils.java

```diff
@@ -0,0 +1,45 @@
+package com.baeldung.capitalizefirstcharactereachword;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.text.WordUtils;
+
+class CapitalizeFirstCharacterEachWordUtils {
+
+    static String usingCharacterToUpperCaseMethod(String input) {
+        if (input == null || input.isEmpty()) {
+            return null;
+        }
+
+        return Arrays.stream(input.split("\\s+"))
+          .map(word -> Character.toUpperCase(word.charAt(0)) + word.substring(1))
+          .collect(Collectors.joining(" "));
+    }
+
+    static String usingStringToUpperCaseMethod(String input) {
+        if (input == null || input.isEmpty()) {
+            return null;
+        }
+
+        return Arrays.stream(input.split("\\s+"))
+          .map(word -> word.substring(0, 1).toUpperCase() + word.substring(1))
+          .collect(Collectors.joining(" "));
+    }
+
+    static String usingStringUtilsClass(String input) {
+        if (input == null || input.isEmpty()) {
+            return null;
+        }
+
+        return Arrays.stream(input.split("\\s+"))
+          .map(StringUtils::capitalize)
+          .collect(Collectors.joining(" "));
+    }
+
+    static String usingWordUtilsClass(String input) {
+        return WordUtils.capitalizeFully(input);
+    }
```

Comment:
We can omit this and just test the WordUtils.capitalizeFully method directly.

Reply:
done